### PR TITLE
changelog for 0.8.0 release

### PR DESCRIPTION
**Pull request by kmike**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -2,6 +2,47 @@
...
+  it consistent with Python's stdlib caching (``lru_cache``,
+  ``cached_property``).
+* The testing framework now uses ``HttpResponse-info.json`` file name instead
+  of ``HttpResponse-other.json`` to store information about HttpResponse
+  instances. To make tests generated with older web-poet work, rename
+  these files on disk.
+
+Testing framework improvements:
+
+* Improved test reporting: better diffs an error messages.
```
Comment (BurnzZ) [inline CHANGELOG.rst:24]:
```suggestion
* Improved test reporting: better diffs on error messages.
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,6 +2,47 @@
...
+
+This release contains many improvements to the web-poet testing framework,
+as well as some other improvements and bug fixes.
+
+Backwards-incompatible changes:
+
+* :func:`~.cached_method` no longer caches exceptions for ``async def`` methods.
+  This makes the behavior the same for sync and async methods, and also makes
+  it consistent with Python's stdlib caching (``lru_cache``,
+  ``cached_property``).
```
Comment (BurnzZ) [inline CHANGELOG.rst:16]:
```suggestion
  it consistent with Python's stdlib caching (i.e. :func:`functools.lru_cache`,
  :func:`functools.cached_property`).
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -2,6 +2,47 @@
...
+  (``--web-poet-test-per-item``) to run a test per item instead.
+* Page objects with the :class:`~.HttpClient` dependency are now supported
+  (see :ref:`web-poet-testing-additional-requests`).
+* Page objects with the :class:`~.PageParams` dependency are now supported.
+* Added a new ``python -m web_poet.testing rerun`` command
+  (see :ref:`web-poet-testing-tdd`).
+* Fixed support for nested (indirect) dependencies in page objects.
+  Previously they were not handled properly by the testing
+  framework.
+* Non-ascii output is now stored without escaping in the test fixtures,
```
Comment (BurnzZ) [inline CHANGELOG.rst:36]: To clarify, would users need to update their test fixtures with this change should they upgrade their web-poet version?

File: CHANGELOG.rst [hunk 4]
```diff
@@ -2,6 +2,47 @@
 Changelog
 =========
 
+0.8.0 (TBD)
+-----------
+
+This release contains many improvements to the web-poet testing framework,
+as well as some other improvements and bug fixes.
+
+Backwards-incompatible changes:
```
Comment (Gallaecio) [inline CHANGELOG.rst:11]:
```suggestion
Backward-incompatible changes:
```

File: CHANGELOG.rst [hunk 5]
```diff
@@ -2,6 +2,47 @@
...
+
+* :func:`~.cached_method` no longer caches exceptions for ``async def`` methods.
+  This makes the behavior the same for sync and async methods, and also makes
+  it consistent with Python's stdlib caching (``lru_cache``,
+  ``cached_property``).
+* The testing framework now uses ``HttpResponse-info.json`` file name instead
```
Comment (Gallaecio) [inline CHANGELOG.rst:17]:
```suggestion
* The testing framework now uses the ``HttpResponse-info.json`` file name instead
```

File: CHANGELOG.rst [hunk 6]
```diff
@@ -2,6 +2,47 @@
...
+  of ``HttpResponse-other.json`` to store information about HttpResponse
+  instances. To make tests generated with older web-poet work, rename
+  these files on disk.
+
+Testing framework improvements:
+
+* Improved test reporting: better diffs an error messages.
+* By default, pytest plugin now generates a test per item attribute
```
Comment (Gallaecio) [inline CHANGELOG.rst:25]:
```suggestion
* By default, the pytest plugin now generates a test per item attribute
```

File: CHANGELOG.rst (hunk 3 above)
Comment (Gallaecio) [inline CHANGELOG.rst:36]:
```suggestion
* Non-ASCII output is now stored without escaping in the test fixtures,
```

Comment (kmike) [reply CHANGELOG.rst:36]: They don't have to; old fixtures are going to work just fine.

**Review by wRAR — APPROVED** [review]

**Review by BurnzZ — APPROVED** [review]

**Pull request merged**